airflow/providers/common/sql/operators/sql.py

```diff
@@ -33,7 +33,21 @@
     from airflow.utils.context import Context
 
 
-def parse_boolean(val: str) -> str | bool:
+def _convert_to_float_if_possible(s):
+    """
+    A small helper function to convert a string to a numeric value
```

Comment:
The code is simple and self explanatory, I think we can skip docs

Reply:
Type hints would however be a good idea here